api/src/main/resources/workbench.yaml

```diff
@@ -967,6 +995,59 @@ definitions:
         format: int64
         description: Milliseconds since the UNIX epoch.
 
+  MaterializeCohortRequest:
+    type: object
+    properties:
+      cohortName:
+        description: >
+         The name of a cohort that is to be evaluated. Either this or cohortSpec should be specified
+        type: string
+      cohortSpec:
+        description: >
+          JSON representation of a cohort to be evaluated (using the same format used for saved
+          cohorts). Either this or cohortName should be specified
+        type: string
+      statusFilter:
+        description: >
+          An array of status values; participants with these statuses will be included.
+          Defaults to [NOT_REVIEWED, INCLUDED, NEEDS_FURTHER_REVIEW] -- everything but EXCLUDED.
+          Only valid for use with cohortName (cohorts saved in the database.)
+        type: array
+        items:
+          $ref: "#/definitions/CohortStatus"
+      cdrVersionName:
+        description: >
+          The name of a CDR version to use when evaluating the cohort; if none is specified,
+          the CDR version currently associated with the workspace will be used
+        type: string
+      token:
+        description: >
+          Pagination token retrieved from a previous call to materializeCohort; used for
+          retrieving additional pages of results. If this is specified, all other fields on
+          MaterializeCohortRequest will be ignored, so don't bother specifying them --
+          the token encodes all the other field values for you.
+        type: string
+      count:
+        description: >
+          Maximum number of results to return. Defaults to 1000.
+        type: int32
+      # TODO: add fieldSpec, output information
+
+  MaterializeCohortResponse:
+    type: object
+    properties:
+      results:
+        description: >
+          An array of JSON dictionaries representing results to the cohort materialization.
+        type: array
+        items:
+          type: object
```

Comment:
Can this be more specific?

Reply:
I don't think so. AFAIK Swagger doesn't support dictionaries.

I need to investigate what this is going to look like in generated TypeScript, Python, etc, and how this gets marshalled... we need some way of reading the dicts out ultimately... I'll get back to you on this.

Reply:
OK - can the description be more specific then? I'm not even sure logically what this is returning - I'd thought it might be participant IDs.

Reply:
My intention was to return an array of dictionaries, one per result row, where the contents of the dictionary would vary based on what you had asked for. So for now:

[ { "participantId": 123 }, { "participantId": 456 } .... ]

and later, other things.

It remains to be seen whether this will  actually work with Swagger codegen; I'll do some experimentation and get back to you on it.

Reply:
OK, updated the doc. I have confirmed that I can add Map<String, Object> entries to the response and parse them out in Typescript. Haven't tried in Python yet, but the generated code looks OK.